instrumentor/controllers/instrumentationdevice/pods_webhook.go

```diff
@@ -28,7 +32,58 @@ func (p *PodsWebhook) Default(ctx context.Context, obj runtime.Object) error {
 		pod.Annotations = map[string]string{}
 	}
 
-	//pod.Annotations["odigos.io/instrumented-webhook"] = "true"
-	log.V(0).Info("Defaulted Pod", "name", pod.Name)
+	// Inject ODIGOS environment variables into all containers
+	injectOdigosEnvVars(pod)
+
 	return nil
 }
+
+func injectOdigosEnvVars(pod *corev1.Pod) {
+
+	namespace := pod.Namespace
+	workloadKind := getWorkloadKind(pod)
+
+	for i := range pod.Spec.Containers {
+		container := &pod.Spec.Containers[i]
+
+		envVars := []corev1.EnvVar{
+			{
+				Name:  EnvVarNamespace,
+				Value: namespace,
+			},
+			{
+				Name:  EnvVarWorkloadKind,
+				Value: workloadKind,
+			},
+			{
+				Name:  EnvVarContainerName,
+				Value: container.Name,
+			},
+			{
+				Name: EnvVarPodName,
+				ValueFrom: &corev1.EnvVarSource{
+					FieldRef: &corev1.ObjectFieldSelector{
+						FieldPath: "metadata.name",
+					},
+				},
+			},
```

Comment:
nit: this can be defined outside of the loop, in each iteration only set the container name